Realm/RLMRealm.mm

```diff
@@ -487,10 +487,21 @@ - (void)removeNotification:(RLMNotificationToken *)token {
 - (void)sendNotifications:(NSString *)notification {
     NSAssert(!_realm->config().read_only, @"Read-only realms do not have notifications");
 
+    NSUInteger count = _notificationHandlers.count;
```

Comment:
I don't understand how these changes improve performance. Seems identical to me, unless the compiler is doing really dumb stuff.

1. the loop won't enter if `count == 0`
2. the loop will run the block exactly once if `count == 1`
3. the loop is identical for `count > 1`

What am I missing?